internal/controller/api/v1alpha1/nmstate_config_types.go

```diff
@@ -20,19 +20,19 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type RawConfig []byte
+type MacInterfaceMapItems struct {
+	// nic name used in the yaml, which relates 1:1 to the mac address
+	LogicalNicName string `json:"logicalNICName"`
+	// mac address present on the host.
+	// Pattern: ^([0-9A-Fa-f]{2}[:]){5}([0-9A-Fa-f]{2})$
```

Comment:
Should use a [validation marker](https://book.kubebuilder.io/reference/markers/crd-validation.html#crd-validation) also to enforce this
```
+kubebuilder:validation:Pattern
```

Reply:
Done